src/Swarm/Game/Step.hs

```diff
@@ -1675,8 +1678,20 @@ updateAvailableRecipes invs e = do
   allInRecipes <- use recipesIn
   let entityRecipes = recipesFor allInRecipes e
       usableRecipes = filter (knowsIngredientsFor invs) entityRecipes
-  knownRecipes <- use availableRecipes
+  Notifications prevCount knownRecipes <- use availableRecipes
   let newRecipes = filter (`notElem` knownRecipes) usableRecipes
       newCount = length newRecipes
-  availableRecipes .= newRecipes <> knownRecipes
-  availableRecipesNewCount += newCount
+  availableRecipes .= Notifications (newCount + prevCount) (newRecipes <> knownRecipes)
+  updateAvailableCommands e
```

Comment:
This looks to me like `Notifications` really wants to be a `Monoid`. 

:thought_balloon:   `availableRecipes <>= Notifications newCount newRecipes`

Reply:
That's tempting, but I think we need another operator to flips the mappend arguments so that the newRecipes are added to the front.

Reply:
Otherwise we can reverse the list when doing the rendering.

Reply:
Oh, right.  I mean, it would still work with something like `deriving Monoid via (Sum Int, Dual [a])` but it's probably not worth it, the current code is fine.  I just really like monoids. 😁

Reply:
In bc73d6f I used `%= mappend` which is a good alternative. I agree, monoids looks much better, thanks!